enterprise_access/apps/api/v1/views.py

```diff
@@ -134,6 +159,119 @@ def _validate_subsidy_request(self):
+                'to approve the requests'
+            )
+            raise SubsidyRequestError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+    @permission_required(
+        constants.REQUESTS_ADMIN_LEARNER_ACCESS_PERMISSION,
+        fn=get_enterprise_uuid_from_request_data,
+    )
+    @action(detail=False, url_path='approve', methods=['post'])
+    def approve(self, *args, **kwargs):
+        """ Action of approving a License Subsidy Request """
+
+        subsidy_request_uuids = self.request.data.get('subsidy_request_uuids')
+        subscription_plan_uuid = self.request.data.get('subscription_plan_uuid')
+        reviewer_lms_user_id = self.lms_user_id
+
+        try:
+            self._validate_subsidy_request_uuids(subsidy_request_uuids)
+            self._validate_subscription_plan_uuid(subscription_plan_uuid)
+            self._verify_subsidies_remaining(subscription_plan_uuid, subsidy_request_uuids)
+        except SubsidyRequestError as exc:
+            logger.exception(exc)
+            return Response(exc.message, exc.http_status_code)
+
+        subsidy_requests = LicenseRequest.objects.filter(uuid__in=subsidy_request_uuids)
+        with transaction.atomic():
+            for subsidy_request in subsidy_requests:
+                if subsidy_request.state == SubsidyRequestStates.DENIED:
+                    error_msg = (
+                        f'License Request with UUID {subsidy_request.uuid} has already been denied. '
+                        'Cannot approve the request.'
+                    )
+                    return Response(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY)
+                subsidy_request.approve(reviewer_lms_user_id)
+
+        # All requests successfully approved, you may now spin off tasks
+        print('SPIN OFF CELERY TASK TO DO THE ASSIGNMENT')
+
+        serialized_subsidy_requests = serializers.LicenseRequestSerializer(subsidy_requests, many=True)
+
+        return Response(
+            serialized_subsidy_requests.data,
+            status=status.HTTP_200_OK,
+        )
+
+    @permission_required(
+        constants.REQUESTS_ADMIN_LEARNER_ACCESS_PERMISSION,
+        fn=get_enterprise_uuid_from_request_data,
+    )
+    @action(detail=False, url_path='deny', methods=['post'])
+    def deny(self, *args, **kwargs):
+        """ Action of denying a License Subsidy Request """
+
+        subsidy_request_uuids = self.request.data.get('subsidy_request_uuids')
+        reviewer_lms_user_id = self.lms_user_id
+
+        try:
+            self._validate_subsidy_request_uuids(subsidy_request_uuids)
+        except SubsidyRequestError as exc:
+            logger.exception(exc)
+            return Response(exc.message, exc.http_status_code)
+
+        subsidy_requests = LicenseRequest.objects.filter(uuid__in=subsidy_request_uuids)
+        with transaction.atomic():
+            for subsidy_request in subsidy_requests:
+                subsidy_request.deny(reviewer_lms_user_id)
```

Comment:
we can't deny approved requests as well I believe so we need to handle that.

Reply:
i guess i figured once something is approved, the UI wouldn't give the option to deny it (and even if someone hit the API directly, it wouldn't change the state of the license, but this is probably the right thing to do, so i'll add it

Reply:
added logic to prevent approved and pending approved requests from being denied 👍